### ISSUE TITLE
ResNet50 and MobileNet performances on ImageNet

### ISSUE THREAD

**Issue by taehoonlee** [keras-team/keras-applications#8]
I have been checked all the models here with TensorFlow backend (see the [commit](https://github.com/keras-team/keras-applications/commit/26c1c11cdf8b2f9b5c3e281c538c1b0fe1a352b3)). For ImageNet validation set, all except ResNet50 and MobileNet showed the same top-k errors with [TensorNets](https://github.com/taehoonlee/tensornets#performances).
- ResNet50: a slight difference (25.296 / 7.980 / 6.852 vs. 25.126 / 7.982 / 6.842).
- MobileNet: a significant difference (34.180 / 13.858 / 10.798 vs. 29.576 / 10.496 / 8.774 for MobileNet(alpha=1.0)).

We need to investigate where the differences come from 🔥.

**Comment by taehoonlee** [comment]
Notes for ResNet50:

| epsilon for BN | ZeroPadding before the first MaxPooling | Performance |
|------|--------------------------|--------------------------|
| 1e-3 | O (112 conv1 -> 56 pool1) | 25.072 / 7.940 (the best) |
| 1e-5 | O (112 conv1 -> 56 pool1) | 25.126 / 7.982 (TensorNets) |
| 1e-3 | X (112 conv1 -> 55 pool1) | 25.296 / 7.980 (Keras) |

- For the epsilon, 1e-3 is the Keras default while 1e-5 is the Caffe default.
- The feature map size of the first pooling is 56x56 with ZeroPadding while 55x55 without ZeroPadding.

**Comment by taehoonlee** [comment]
Notes for MobileNet:
- There are two `ZeroPadding2D((1, 1))`s in both `_conv_block` and `_depthwise_conv_block`.
- With the two, all the backends can produce the same results, which are poor (34.180 / 13.858 / 10.798).
- Without the two, TensorNets and Keras with TF backend can produce the same and good results (29.576 / 10.496 / 8.774); but Theano and CNTK backends resulted in different results.
